packages/sanity/src/core/releases/tool/components/releaseCTAButtons/ReleaseRevertButton/useDocumentRevertStates.ts

```diff
@@ -0,0 +1,135 @@
+import {type SanityDocument} from '@sanity/types'
+import {useCallback, useEffect, useMemo, useRef} from 'react'
+import {useObservable} from 'react-rx'
+import {catchError, filter, forkJoin, from, map, type Observable, of, switchMap} from 'rxjs'
+
+import {useClient} from '../../../../../hooks/useClient'
+import {getTransactionsLogs} from '../../../../../store/translog/getTransactionLogs'
+import {API_VERSION} from '../../../../../tasks/constants'
+import {type DocumentInRelease} from '../../../detail/useBundleDocuments'
+
+export type RevertDocument = SanityDocument & {
+  _system?: {
+    delete: boolean
```

Comment:
Documents with a `_system.delete` property would never have `delete: false`, right? So this could be typed more narrowly to `delete: true`, or?

Reply:
Good idea!